tests/equality/equality.cpp

```diff
@@ -0,0 +1,995 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "gtest/gtest.h"
+
+#include <libcellml>
+
+#include "test_utils.h"
+
+TEST(Equality, invalidInput)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
```

Comment:
auto?  also throughout

Reply:
We don't have to use `auto` just because we can.  For me differences in the tests are a good thing.